pandas/core/indexes/numeric.py

```diff
@@ -157,10 +156,13 @@ def _ensure_array(cls, data, dtype, copy: bool):
         if issubclass(data.dtype.type, str):
             cls._string_data_error(data)
 
-        dtype = cls._ensure_dtype(dtype)
+        dtype = cls._ensure_dtype(dtype, validate=False)
 
         if copy or not is_dtype_equal(data.dtype, dtype):
             subarr = np.array(data, dtype=dtype, copy=copy)
+            if not is_numeric_dtype(subarr.dtype):
+                # hack to raise correctly
```

Comment:
im not clear what "raise correctly" means here

Reply:
i agree, why is this needed?

Reply:
I've looked at it again, it's a bit tricky, because dtype=None and dtype being an numpy dtype will give different error messages.

I'll look again at it later.

Reply:
Still tricky.

Can we agree that Float64Index(arr, dtype=""int64") should raise a TypeError, not a ValueError as it currently does. That might simplify a solution.

Reply:
The newest version has a new implementation.